packages/rabbitmq/test/rabbitmqTxSubmitProvider.test.ts

```diff
@@ -0,0 +1,54 @@
+import { ProviderError, TxSubmitProvider } from '@cardano-sdk/core';
+import { RabbitMqTxSubmitProvider } from '../src';
+
+const BAD_CONNECTION_URL = new URL('amqp://localhost:1234');
+const GOOD_CONNECTION_URL = new URL('amqp://localhost');
+
+describe('RabbitMqTxSubmitProvider', () => {
+  let provider: TxSubmitProvider;
+
+  afterEach(() => provider?.close!());
+
+  describe('healthCheck', () => {
+    it('is not ok if cannot connect', async () => {
+      provider = new RabbitMqTxSubmitProvider(BAD_CONNECTION_URL);
+      const res = await provider.healthCheck();
+      expect(res).toEqual({ ok: false });
+    });
+
+    it('is ok if cannot connect', async () => {
+      provider = new RabbitMqTxSubmitProvider(GOOD_CONNECTION_URL);
+      // Called twice to cover AMQPWrapper.#connect() first line as well
+      const resA = await provider.healthCheck();
+      const resB = await provider.healthCheck();
+      expect(resA).toEqual({ ok: true });
+      expect(resB).toEqual({ ok: true });
+    });
+  });
+
+  const performSubmitTxTest = async (connectionURL: URL) => {
+    try {
+      provider = new RabbitMqTxSubmitProvider(connectionURL);
+      // Called twice to cover AMQPWrapper.#createQueue() first line as well
+      const resA = await provider.submitTx(new Uint8Array());
+      const resB = await provider.submitTx(new Uint8Array());
+      expect(resA).toBeUndefined();
+      expect(resB).toBeUndefined();
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      expect(error.innerError).toBeInstanceOf(ProviderError);
+    }
+  };
+
+  describe('submitTx', () => {
+    it('resolves if successful', async () => {
+      await performSubmitTxTest(GOOD_CONNECTION_URL);
+      expect.assertions(2);
```

Comment:
Best to make the assertion at the top of the function block (line 45)

Reply:
Do you mean swapping line 45 with line 46?

Reply:
Yes, so that we're making the expectation, then executing the code to trigger the assertions.